samples/memc/stm32_backup_sram/src/main.c

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+
+/** Value stored in backup SRAM. */
+__stm32_backup_sram_section uint32_t backup_value;
+
+void main(void)
+{
+	printk("Current value in backup SRAM: %d\n", backup_value);
+
+	backup_value++;
+
+	printk("Next reported value should be: %d\n", backup_value);
+	printk("Re-power the board now (excepting VBAT)\n");
+
+#if __DCACHE_PRESENT
+	SCB_CleanDCache_by_Addr(&backup_value, sizeof(backup_value));
```

Comment:
Seems to be required on M7 (see https://community.st.com/s/question/0D50X0000ALwOvkSQF/i-can-not-use-backup-ram-for-stm32h743bi-is-there-any-application-note-or-source-code-for-backup-ram). Not sure if there is a better/more generic solution.